clients/rust/tests/update_external_plugins_on_collection.rs

```diff
@@ -0,0 +1,292 @@
+                )],
+                base_address_config: None,
+                results_offset: None,
+            })],
+        },
+    )
+    .await
+    .unwrap();
+
+    let update_authority = context.payer.pubkey();
+    assert_collection(
+        &mut context,
+        AssertCollectionHelperArgs {
+            collection: collection.pubkey(),
+            update_authority,
+            name: None,
+            uri: None,
+            num_minted: 0,
+            current_size: 0,
+            plugins: vec![],
+            external_plugin_adapters: vec![ExternalPluginAdapter::Oracle(Oracle {
+                base_address: Pubkey::default(),
+                base_address_config: None,
+                results_offset: ValidationResultsOffset::NoOffset,
+            })],
+        },
+    )
+    .await;
+
+    let ix = UpdateCollectionExternalPluginAdapterV1Builder::new()
+        .collection(collection.pubkey())
+        .payer(context.payer.pubkey())
+        .key(ExternalPluginAdapterKey::Oracle(Pubkey::default()))
+        .update_info(ExternalPluginAdapterUpdateInfo::Oracle(OracleUpdateInfo {
+            lifecycle_checks: Some(vec![
+                (
+                    HookableLifecycleEvent::Transfer,
+                    ExternalCheckResult { flags: 4 },
+                ),
+                (
+                    HookableLifecycleEvent::Transfer,
+                    ExternalCheckResult { flags: 4 },
+                ),
+            ]),
+            base_address_config: None,
+            results_offset: Some(ValidationResultsOffset::Custom(10)),
+        }))
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let error = context
+        .banks_client
+        .process_transaction(tx)
+        .await
+        .unwrap_err();
+
+    assert_custom_instruction_error!(0, error, MplCoreError::DuplicateLifecycleChecks);
+
+    // TODO: Check that registry record did not get updated.
+}
+
+#[tokio::test]
+async fn test_update_app_data_on_collection() {
+    let mut context = program_test().start_with_context().await;
+
+    let collection = Keypair::new();
+    create_collection(
+        &mut context,
+        CreateCollectionHelperArgs {
+            collection: &collection,
+            update_authority: None,
+            payer: None,
+            name: None,
+            uri: None,
+            plugins: vec![],
+            external_plugin_adapters: vec![ExternalPluginAdapterInitInfo::AppData(
+                AppDataInitInfo {
+                    init_plugin_authority: Some(PluginAuthority::UpdateAuthority),
+                    data_authority: PluginAuthority::UpdateAuthority,
+                    schema: None,
+                },
+            )],
+        },
+    )
+    .await
+    .unwrap();
+
+    let update_authority = context.payer.pubkey();
```

Comment:
I feel like we talked about this but I don't remember the details. Does it make sense for the UA to control the schema when the data auth is the one managing it?

Reply:
Talked offline.  In follow on PR I will look into using the plugin authority to control the schema changes